src/ipred.rs

```diff
@@ -930,10 +933,18 @@ unsafe fn ipred_z2_rust<BD: BitDepth>(
     } else {
         false
     };
-    let mut edge: [BD::Pixel; 129] = [0.into(); 129];
-    let topleft: *mut BD::Pixel = &mut *edge.as_mut_ptr().offset(64) as *mut BD::Pixel;
+    let mut edge = [0.into(); 129];
```

Comment:
```suggestion
    let mut edge = [0.into(); 64 + 64 + 1];
```

Reply:
This PR isn't focused on cleaning up the `ipred_z*_rust` functions, I'm going to do that in a separate PR.